### PR TITLE
Added jsdoc comments

### PR DESCRIPTION
**Pull request by BernsteinA**
This shows up in hover tooltips in many text editors, including vs code

### CODE REVIEW

**Comment by BernsteinA** [comment]
ping @clark-stevenson is this the right place to make this PR, to get the changes into DefinitelyTyped?

**Comment by ivanpopelyshev** [comment]
Yep, its the right place for pixi-v4 typings.

**Comment by BernsteinA** [comment]
Great, thanks

**Comment by clark-stevenson** [comment]
@BernsteinA 

Sorry! I was on Vacation for a few weeks but back! 

I appreciate your PR but there is an issue with this Repo in that it is unrelated to DefinitelyTyped directly so it is a manual process to then copy everything that happens here into a new PR over there and the files aren't compatible. 

It has always been a mess, but the v4 stuff is kind of somewhat unchanging and to an extent, I consider the types somewhat sunset barring major API changes or additions.    It will be a major slog to go through all of the DT bureaucracy  and manually document everything.   Effectively taking almost as long as the declaration itself. 


So I don't know where to go with this guys,  can you give me any input?  The dozens of hours it will take to fully complete this PR in 2 locations, is time which I believe is better spent improving v5 for TYpeScript Developers tomorrow.  To a certain extent,  I feel we are polishing a dead body if we continue?

**Comment by BernsteinA** [comment]
Let me provide a bit more context.
I just started a new job, where I use pixi v4 with typescript on a daily basis, on multiple projects. We're not likely to upgrade to v5.
These changes represent a real upgrade to my workflow. I know that because I'm using them in my local repos.
If it's too much hassle to push this all the way upstream, I get it.

**Comment by clark-stevenson** [comment]
@BernsteinA  

Do you use Definitely Typed declarations via `npm install @types/pixi.js` in your day to day dev?

@ivanpopelyshev 

I am going to presume that almost everyone these days uses `npm install @types/pixi.js` for v4?  



What would happen if I went over to Definitely Typed, copied the definition, and then pasted it in here?  Like a full replacement?  Would that be an OK thing to do?   Really the only difference between them is in the syntax and structure since MS imposes heavy TSLinting rules that this repo never had. 


In other words, I believe the DT file today is far superior, but I don't know if I break anyone using this repo by doing that.


Would love any thoughts.

**Comment by BernsteinA** [comment]
Yup, we use Definitely Typed via @types/pixi.js

I can do the labor of reformatting the PR if it helps. The type definitions
I was able to generate automatically didn't match the ones in this repo, so
my work this far has been rather manual already.

**Comment by clark-stevenson** [comment]
@BernsteinA 

Awesome to know!  Also I really appreciate you taking the time for all this.  It was a fair amount of changes. 

My plan is to go ahead with issue #224   

To do that, I will take this PR on board.  I have already started a new DT PR and I have manually copied your work over to that.   Then, once I get a green light on DT,  I will take that file and put it in this repo.    

I will keep you in the loop, more to follow....

**Comment by clark-stevenson** [comment]
Awaiting Merge on https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38246

**Comment by BernsteinA** [comment]
Thank you!
